src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsOnSpecialOasis.java

```diff
@@ -0,0 +1,79 @@
+package com.wikia.webdriver.testcases.adstests;
+
+import com.wikia.webdriver.common.templates.TemplateNoFirstLoad;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsBaseObject;
+
+import org.openqa.selenium.Dimension;
+import org.testng.annotations.Test;
+
+/**
+ * https://www.google.com/dfp/5441#delivery/LineItemDetail/lineItemId=126608052
+ */
+
+public class TestAdsOnSpecialOasis extends TemplateNoFirstLoad {
+
+  @Test(groups = "TestAdsOnSpecialPagesOasis")
+  public void testAdsOnSpecialVideoPageOasis() throws Exception {
+    AdsBaseObject ads = buildAdsObjectForPage("Special:Videos");
+    verifySlotHasCorrectAd(
+        ads,
+        "TOP_LEADERBOARD",
+        "wka.life/_project43//special",
+        "271491732"
+    );
+  }
+
+  @Test(groups = "TestAdsOnSpecialPagesOasis")
+  public void testAdsOnSpecialImagesPageOasis() throws Exception {
+    AdsBaseObject ads = buildAdsObjectForPage("Special:Images");
+    verifySlotHasCorrectAd(
+        ads,
+        "TOP_LEADERBOARD",
+        "wka.life/_project43//special",
+        "271491732");
+  }
+
+  private AdsBaseObject buildAdsObjectForPage(String pageName) {
+    String testedPage = urlBuilder.getUrlForPath("project43", pageName);
+    return new AdsBaseObject(driver, testedPage, new Dimension(1292, 1000));
+  }
+
+  private void verifySlotHasCorrectAd(
+      AdsBaseObject ads,
+      String slotName,
+      String adUnit,
+      String lineItem
+  ) {
+    ads.verifyGptIframe(adUnit, slotName, "gpt");
+    ads.verifyGptAdInSlot(slotName, lineItem, "");
+  }
+
+  @Test(groups = "TestAdsOnFilePagesOasis")
+  public void testAdsOnFilePageOasis() throws Exception {
+    AdsBaseObject ads = buildAdsObjectForPage("File:Example.jpg");
+    String adUnit = "wka.life/_project43//file";
+
+    verifySlotHasCorrectAd(
```

Comment:
L56-L76 - it seems the slot name is only difference. Can we define some array with slot names and verify slots in loop (it will require reusing SLOT_TRIGGERS in case of BLB.

Reply:
I think it makes sense. We could have "per layout" `SLOTS_SELECTORS` for example:
* `PAL_SLOTS_SELECTORS` - premium ad layout slots' selectors,
* `SLOTS_SELECTORS` - old ad layout slots' selectors,
* `FILE_PAGE_SELECTORS` etc.
and the same `SLOTS_TRIGGERS`. 

If I do it in this PR I'll introduce conflicts and there are some changes already in other PRs. Let's make it in a separate step: [ADEN-6015](https://wikia-inc.atlassian.net/browse/ADEN-6015).

Reply:
👍